src/main/proto/netflix/titus/titus_vpc_api.proto

```diff
@@ -50,3 +51,117 @@ message ValidationResponse {
 service ValidatorIPService {
     rpc ValidateAllocation (ValidationRequest) returns (ValidationResponse);
 }
+
+// V2 API
+message AllocateStaticIPAddressRequest {
+    /// Required (The subnet where to allocate the address)
+    string subnetId = 1;
+    /// Optional
+    string uuid = 2;
+    /// Optional
+    string pool = 3;
+    /// Optional
+    map<string, string> tags = 4;
+};
+
+message AllocateStaticIPAddressResponse {
+    StaticIPAddress staticIPAddress = 1;
+};
+
+message AddressLocationV2 {
+    string accountId = 1;
+    string availabilityZone = 2;
+    string availabilityZoneId = 3;
+    string vpcId = 4;
+    string subnetId = 5;
+};
+
+message StaticIPAddress {
+    string uuid = 1;
+    string ipv4Address = 2;
+    string ipv6Address = 3;
+    string pool = 4;
+    map<string, string> tags = 5;
+    AddressLocationV2 addressLocation = 6;
+};
+
+message SetPoolRequest {
+    string uuid = 1;
+    // If Pool is unset, it's equivalent of removing it from that pool
+    string pool = 2;
+};
+
+message SetPoolResponse {
+};
+
+message CreateTagRequest {
+    string uuid = 1;
+    string key = 2;
+    string value = 3;
+    // If upsert is false, and the tag exists, this call will fail.
+    bool upsert = 4;
+};
+
+message CreateTagResponse {
+};
+
+message DeleteTagRequest {
+    string uuid = 1;
+    string key = 2;
+};
+
+message DeleteTagResponse {
+};
+
+// If the tag is unset, then do not set it
+message UpdateTagRequest {
+    string uuid = 1;
+    string key = 2;
+    string value = 3;
+};
+
+message UpdateTagResponse {
+};
+
+message AddressSearchParameter {
+    string address = 1;
+    oneof location {
+        // This can be empty if address is a v6 address, otherwise it must be specified
+        google.protobuf.Empty empty = 2;
```

Comment:
Couldn't a user just leave both subnetId and vpcId unset and not need the Empty? Also, why account or AZ?

Reply:
You mean, if this is wasn't a `oneof`? If you have multiple VPCs per account, you can overlapping space. IPs are only guaranteed to be unique within a VPC. Subnets are bound to a VPC.